crates/sui-framework/packages/deepbook/sources/clob_v2.move

```diff
@@ -1317,6 +1317,55 @@ module deepbook::clob_v2 {
         }
     }
 
+    /// Clean up expired orders
+    /// Note that this function can reduce gas cost if orders
+    /// with the same price are grouped together in the vector because we would not need the computation to find the tick_index.
+    /// For example, if we have the following order_id to price mapping, {0: 100., 1: 200., 2: 100., 3: 200.}.
+    /// Grouping order_ids like [0, 2, 1, 3] would make it the most gas efficient.
+    /// Order owners should be the owner addresses from the account capacities which placed the orders,
+    /// and they should correspond to the order IDs one by one.
+    public fun clean_up_expired_orders<BaseAsset, QuoteAsset>(
+        pool: &mut Pool<BaseAsset, QuoteAsset>,
+        clock: &Clock,
+        order_ids: vector<u64>,
+        order_owners: vector<address>
+    ) {
+        let pool_id = *object::uid_as_inner(&pool.id);
+        let now = clock::timestamp_ms(clock);
+        let n_order = vector::length(&order_ids);
+        assert!(n_order == vector::length(&order_owners), ENotEqual);
+        let i_order = 0;
+        let tick_index: u64 = 0;
+        let tick_price: u64 = 0;
+        while (i_order < n_order) {
+            let order_id = *vector::borrow(&order_ids, i_order);
+            let owner = *vector::borrow(&order_owners, i_order);
+            let usr_open_orders = borrow_mut(&mut pool.usr_open_orders, owner);
+            let new_tick_price = *linked_table::borrow(usr_open_orders, order_id);
```

Comment:
Similarly, this will crash if `order_id` has already been cancelled. Because anyone can call this function, I think there will frequently be races to cancel the same order, and it will be frustrating if losing a race makes your other cancellations fail. 

I think it would be better to check if `order_id` exists and `continue` if it does not--this will allow partial success in cancelling orders.